File: app/src/main/java/de/fhswf/fahrradverleih/android/view/tabs/viewmodel/TabsViewModel.java
package de.fhswf.fahrradverleih.android.view.tabs.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

/**
 * Verwaltet, welcher Tab angezeigt wird, und verwaltet Navigationsevents.
 */
public class TabsViewModel extends ViewModel {

    private final MutableLiveData<TabsContent> content =
            new MutableLiveData<>(TabsContent.MAP);

    public MutableLiveData<TabsContent> getContent() {
        return content;
    }

    public void setContent(@NonNull TabsContent content) {
        if (getContentValue() != content)
            this.content.setValue(content);
    }

    @NonNull
    public TabsContent getContentValue() {
        if (content.getValue() == null) throw new IllegalStateException();
        return content.getValue();
    }

    /**
     * Verarbeitet Navigationsevents.
     *
     * @return true -> Activity beenden ; false -> Standardverhalten unterdrücken
     */
    public boolean backPressed() {
        if (getContentValue() != TabsContent.MAP) {
            setContent(TabsContent.MAP);
            return false;
        } else return true;
    }
}

File: app/src/main/java/de/fhswf/fahrradverleih/android/view/tabs/viewmodel/TabsViewModel.kt
package de.fhswf.fahrradverleih.android.view.tabs.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

enum class TabsContent {
    MAP, AUSLEIHEN, PROFIL
}

class TabsViewModel : ViewModel() {

    private val content = MutableLiveData(TabsContent.MAP)

    fun getContent(): LiveData<TabsContent> = content

    fun navigateTo(content: TabsContent) {
        this.content.value = content
    }

    fun backPressed(): Boolean {
        return this.content.value == TabsContent.MAP
    }

}